drivers/scheduler/spec/spec.go

```diff
@@ -14,3 +14,18 @@ type AppSpec struct {
 	// Enabled indicates if the application is enabled in the factory
 	Enabled bool
 }
+
+// DeepCopy Creates a copy of the AppSpec
+func (in *AppSpec) DeepCopy() *AppSpec {
+	if in == nil {
+		return nil
+	}
+	out := new(AppSpec)
+	out.Key = in.Key
+	out.Enabled = in.Enabled
+	out.SpecList = make([]interface{}, len(in.SpecList))
+	for _, spec := range in.SpecList {
+		out.SpecList = append(out.SpecList, spec)
+	}
```

Comment:
the spec if a pointer will still point to the same object, right?
most of our specs are pointers

Reply:
Changed it to use a library which should do it correctly